pyfa_tool/dataset.py

```diff
@@ -0,0 +1,797 @@
+            is None.
+        title : str, optional
+            The title of the plot. If None, a default title is constructed. The
+            default is None.
+        grid : bool, optional
+            Add gridlines to plot. The default is False.
+        land : bool, optional
+            If True, and if the dataset is in a latlon projection (EPSG:4326),
+            then land boarders are drawn. If None, it will be set to True if
+            the projection is latlon, else False. The default is None.
+        coastline : bool, optional
+            If True, and if the dataset is in a latlon projection (EPSG:4326),
+            then coastlines are drawn. If None, it will be set to True if
+            the projection is latlon, else False. The default is None.
+        contour : bool, optional
+            If True, the contourf() method is used as a plotting backend, else
+            the default xarray.Dataset.plot(). The default is False.
+        contour_levels : int, optional
+            Number of contour levels to use, if contour=True. The default is 10.
+        **kwargs : kwargs
+            Kwargs will be passed to the xarray.plot() method.
+
+        Returns
+        -------
+        ax : matplotlib.pyplot.axes
+            The geospatial axes of the plot.
+
+        """
+
+
+
+        assert not (self.ds is None), 'Empty instance of FaDataset.'
+
+        if self._is_3d_field(variable):
+            assert not (level is None), f'{variable} is a 3D field. Specify a level.'
+            xarr = self.ds[variable].sel(level=level)
+        elif self._is_2d_field(variable):
+            xarr = self.ds[variable]
+        else:
+            sys.exit('something unforseen is wrong.')
+
+        # setup default values for coastline and land features
+        islatlon = self. _in_latlon()
+        if land is None:
+            if islatlon:
+                land=True
+            else:
+                land=False
+
+        if coastline is None:
+             if islatlon:
+                 coastline=True
+             else:
+                 coastline=False
+
+        if (land) | (coastline):
+            if not islatlon:
+                sys.exit('Adding land and coastline features is only available in latlon coordinates')
+
+        if islatlon:
+            fig, ax = plotting.make_platcarree_fig()
+        else:
+            fig, ax = plotting.make_regular_fig()
+
+        # create title
+        if title is None:
+            title =  title=f'{variable} at {self.ds.attrs["validate"]} (UTC, LT={self.ds.attrs["leadtime"]}h)'
+
+
+        ax = plotting.make_plot(dxr = xarr,
+                        ax=ax,
+                        title=title,
+                        grid=grid,
+                        land=land,
+                        coastline=coastline,
+                        contour=contour,
+                        levels=contour_levels,
+                        **kwargs)
+
+        return ax
+
+
+
+
+    def describe(self):
```

Comment:
Also not essential but may be nice to describe both what is in the FA file (by linking/making a Fa File object and what is currently in the ds variable (i.e. what has already been converted to xarray).

Reply:
convert to issue #46